File: questions/BinarySearch.py
arr = [1,2,3,4,5,6,7,8,9]
def search(arr, x, high, low):
    mid = low + (high - low) // 2
    if x == arr[mid]:
        return mid
    else:
        if x > arr[mid]:
            return search(arr, x, high, mid+1)
        else:
            return search(arr, x, mid-1, low)

print(search(arr, 8, len(arr)-1, 0))

File: Lab 2.0/BinarySearch.java
import java.util.Scanner;

public class BinarySearch {
    public static void main(String[] args) {
        int size;
        // take input for number of elements
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of elements: ");
        size = sc.nextInt();
        // create an array of size 'size'
        int[] arr = new int[size];
        // take input for elements
        System.out.print("Enter " + size + " elements: ");
        for (int i = 0; i < size; i++) {
            arr[i] = sc.nextInt();
        }
        // take input for target
        System.out.print("Enter the target: ");
        int target = sc.nextInt();
        sc.close();
        // call binarySearch() and store the returned value in ans
        int ans = binarySearch(arr, target);
        // print ans
        System.out.println("Target found at position: "+(ans+1));
    }

    private static int binarySearch(int[] arr, int target) {
        int start = 0;
        int end = arr.length - 1;
        while (start <= end) {
            int mid = (start + end) / 2;
            if (target < arr[mid]) {
                end = mid - 1;
            } else if (target > arr[mid]) {
                start = mid + 1;
            } else {
                return mid;
            }
        }
        return -1;
    }
}
